File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/MobileConfigurations/CardControl/LostOrStolenCardSettings.cs
using System.Collections.Generic;

namespace Psi.Data.Models.ClientConfigurationModels.MobileConfigurations.CardControl
{
    public class LostOrStolenCardSettings : SettingsBaseHelper
    {
        public LostOrStolenCardSettings(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("Mobile.CardControl.LostOrStolen.Enabled")]
        public bool Enabled
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("Mobile.CardControl.LostOrStolen.SendSecureMessageEnabled")]
        public bool SendSecureMessageEnabled
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("Mobile.CardControl.LostOrStolen.EligibleCardTypes")]
        public List<string> EligibleCardTypes
        {
            get => GetListValue();
            set => SetValue(value);
        }

        [SettingKey("Mobile.CardControl.LostOrStolen.Locations")]
        public List<string> Locations
        {
            get => GetListValue();
            set => SetValue(value);
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/CardControl/LostOrStolenCardSettings.ts
// Generated imports
import { EligibleCardTypes } from '../EligibleCardTypes';
import { Locations } from '../Locations';

export interface LostOrStolenCardSettings {
    /** @settingKey Mobile.CardControl.LostOrStolen.Enabled */
    enabled: boolean;
    /** @settingKey Mobile.CardControl.LostOrStolen.SendSecureMessageEnabled */
    sendSecureMessageEnabled: boolean;
    /** @settingKey Mobile.CardControl.LostOrStolen.EligibleCardTypes */
    list: EligibleCardTypes;
    /** @settingKey Mobile.CardControl.LostOrStolen.Locations */
    list: Locations;
}
